app/src/main/java/org/jellyfin/androidtv/ui/browsing/DisplayPreferencesScreen.kt

```diff
@@ -46,6 +47,19 @@ class DisplayPreferencesScreen : OptionsFragment() {
 					default { "0" }
 				}
 			}
+			list {
+				setTitle(R.string.grid_direction)
+				entries = mapOf(
+					GridDirection.HORIZONTAL.name to requireContext().getString(R.string.grid_direction_horizontal),
+					GridDirection.VERTICAL.name to requireContext().getString(R.string.grid_direction_vertical),
+				)
+
+				bind {
+					get { displayPreferences.getCustomPrefs().get("GridDirection") ?: GridDirection.HORIZONTAL.name }
+					set { displayPreferences.getCustomPrefs().set("GridDirection", it) }
+					default { GridDirection.HORIZONTAL.name }
+				}
+			}
```

Comment:
There is an `enum` function that uses a list in the back. Something like this should work, although this sample can cause a crash if the value is not null and an bad value.

```suggestion
			enum<GridDirection> {
				setTitle(R.string.grid_direction)

				bind {
					get { displayPreferences.getCustomPrefs().get("GridDirection")?.let(GridDirection::valueOf) ?: GridDirection.HORIZONTAL }
					set { displayPreferences.getCustomPrefs().set("GridDirection", it.name) }
					default { GridDirection.HORIZONTAL }
				}
			}
```